pyiceberg/io/pyarrow.py

```diff
@@ -1772,12 +1772,13 @@ def write_file(io: FileIO, table_metadata: TableMetadata, tasks: Iterator[WriteT
     )
 
     def write_parquet(task: WriteTask) -> DataFile:
+        df = pa.Table.from_batches(task.record_batches)
+        df = df.rename_columns(schema.column_names)
```

Comment:
we need to change the Schema (column names) of the arrow data frame, if there's a better way to do this, please let me know

Reply:
Shall we extend the integration test to test the nested schema case? For example, 
```python
        pa.field('name', pa.string()),
        pa.field('address', pa.struct([
            pa.field('street', pa.string()),
            pa.field('city', pa.string()),
            pa.field('zip', pa.int32())
        ]
```

Updated: I got 
```python
pyarrow.lib.ArrowInvalid: tried to rename a table of 4 columns but only 7 names were provided
```
when trying with the following dataset
```python
TEST_DATA_WITH_SPECIAL_CHARACTER_COLUMN = {
        column_name_with_special_character: ['a', None, 'z'],
        'id': [1, 2, 3],
        'name': ['AB', 'CD', 'EF'],
        'address': [
            {'street': '123', 'city': 'SFO', 'zip': 12345},
            {'street': '456', 'city': 'SW', 'zip': 67890},
            {'street': '789', 'city': 'Random', 'zip': 10112}
        ]
    }
    pa_schema = pa.schema([
        pa.field(column_name_with_special_character, pa.string()),
        pa.field('id', pa.int32()),
        pa.field('name', pa.string()),
        pa.field('address', pa.struct([
            pa.field('street', pa.string()),
            pa.field('city', pa.string()),
            pa.field('zip', pa.int32())
        ]))
    ])
```

Reply:
good catch! the i dont think `rename_columns` works well with nested schema